docs/sphinx/source/conf.py

```diff
@@ -84,4 +85,98 @@
 master_doc = 'index'
 # A workaround for the responsive tables always having annoying scrollbars.
 def setup(app):
-    app.add_stylesheet("no_scrollbars.css")
\ No newline at end of file
+    app.add_stylesheet("no_scrollbars.css")
+
+
+# %% helper functions for intelligent "View on Github" linking
+# based on
+# https://gist.github.com/flying-sheep/b65875c0ce965fbdd1d9e5d0b9851ef1
+
+def get_obj_module(qualname):
+    """
+    Get a module/class/attribute and its original module by qualname.
+    Useful for looking up the original location when a function is imported
+    into an __init__.py
+
+    Examples
+    --------
+    >>> func, mod = get_obj_module("rdtools.degradation_ols")
+    >>> mod.__name__
+    'rdtools.degradation'
+    """
+    modname = qualname
+    classname = None
+    attrname = None
+    while modname not in sys.modules:
+        attrname = classname
+        modname, classname = modname.rsplit('.', 1)
+
+    # retrieve object and find original module name
+    if classname:
+        cls = getattr(sys.modules[modname], classname)
+        modname = cls.__module__
+        obj = getattr(cls, attrname) if attrname else cls
+    else:
+        obj = None
+
+    return obj, sys.modules[modname]
+
+
+def get_linenos(obj):
+    """Get an object’s line numbers in its source code file"""
+    try:
+        lines, start = inspect.getsourcelines(obj)
+    except TypeError:  # obj is an attribute or None
+        return None, None
+    else:
+        return start, start + len(lines) - 1
+
+
+def make_github_url(pagename):
+    """
+    Generate the appropriate GH link for a given docs page.  This function
+    is intended for use in sphinx template files.
+    The target URL is built differently based on the type of page.  Sphinx
+    provides templates with a built-in `pagename` variable that is the path
+    at the end of the URL, without the extension.  For instance,
+    https://rdtools.rtfd.io/en/latest/generated/rdtools.soiling.soiling_srr.html
+    will have pagename = "generated/rdtools.soiling.soiling_srr".
+    """
+
+    # RTD automatically sets READTHEDOCS_VERSION to the version being built.
+    if os.environ.get('READTHEDOCS_VERSION', None) == 'stable':
+        branch = 'master'
+    else:
```

Comment:
Let's  be explicit about all the cases here. Otherwise, if we change the readthedocs configuration, we could end up with links pointing to development when they shouldn't. Not sure what the correct behavior should be for PR and local builds though. I'd almost rather it be broken than point to the wrong branch.

Reply:
Handling PR build correctly seems like a huge headache for not much benefit, and I'm not even sure what a local build should point to.  What do you think about only generating the link if it's stable/master or latest/development?  For other builds it would just omit that HTML element.